src/Model/Lumberjack.php

```diff
@@ -65,57 +68,77 @@ public function updateCMSFields(FieldList $fields)
     }
 
     /**
-     * Augments (@link Hierarchy::stageChildren()}
+     * Return children in the stage site.
      *
-     * @param boolean $showAll Include all of the elements, even those not shown in the menus.
-     *   (only applicable when extension is applied to {@link SiteTree}).
+     * @param bool $showAll Include all of the elements, even those not shown in the menus. Only applicable when
+     *                      extension is applied to {@link SiteTree}.
      * @return DataList
      */
     public function stageChildren($showAll = false)
     {
-        $baseClass = ClassInfo::baseDataClass($this->owner->class);
-        $staged = $baseClass::get()
+        $hideFromHierarchy = $this->owner->config()->hide_from_hierarchy;
```

Comment:
As you address the config multiple times, I would go with 
```
$config = $this->owner->config();
$hideFromHierarchy = $config->get('hide_from_hierarchy');
$hideFromCMSTree = $config->get('hide_from_cms_tree');
```
which (I feel) is more readable.